### PR TITLE
[Relax] Handle binary operations between Tensor and PrimValue

### PR DESCRIPTION
**Pull request by Lunderberg**
Prior to this commit, binary operations were only defined between two tensors.  This commit allows binary operations to apply between a tensor and a `relax::PrimValue`.

When inferring the output `StructInfo`, binary operations with a `PrimValue` produce the same output as using a 0-d tensor.  When legalizing operations containing a `PrimValue`, they are lowered to primitive TIR arguments.

### CODE REVIEW

File: src/script/printer/relax/tir.cc
Comment (Lunderberg) [reply src/script/printer/relax/tir.cc:46]: Good point.  I needed to remove the `n->dtype.is_int()` check, as the value could be `R.Prim("float32")`, but the check for `n->dtype.is_scalar()` should be kept.

File: src/relax/op/op_common.h
Comment (slyubomirsky) [reply src/relax/op/op_common.h]: Personally, I'm in favor of asking for a MatchCast if we can't draw a conclusion. Down the line, inserting MatchCasts via normalization rules would be a good policy.

**Review by slyubomirsky — APPROVED** [review]
Thanks for making the requested changes. I don't see much harm in adding support for such ops, though we should be mindful of the added complexity of having more ways to express equivalent computations. I don't think it's a problem for now.

File: src/relax/op/op_common.h
Comment (Lunderberg) [reply src/relax/op/op_common.h]: True.  Currently, `FInferStructInfo` is called prior to `FNormalize`, so inference could be inspecting an expression that hasn't yet been normalized.  This was useful for providing `FNormalize` for `R.Prim` (if `PrimStructInfo` contains a known value, in-line that value), but I'm wondering if we should re-visit that.

File: tests/python/relax/test_op_binary.py
Comment (Lunderberg) [reply tests/python/relax/test_op_binary.py]: Thank you for the catch, and they are now uncommented.

**Comment by Lunderberg** [comment]
@tqchen The requested change has been made, and CI is passing.  Any other changes that should be made before merging?

**Review by tqchen — APPROVED** [review]

File: src/relax/op/op_common.h
Comment (tqchen) [inline src/relax/op/op_common.h:255]: Originally our error message would ask for TensorStructInfo. In this particular case, would this error message be less informative than before? Given this is a global change across all binary ops, would be good to cross confirm the usages here and make error more informative.

**Review by tqchen — CHANGES_REQUESTED** [review]

**Comment by tqchen** [comment]
Sorry i didn't yet have time to do a full look through, will spend sometime this weekend

File: src/relax/op/op_common.h
Comment (Lunderberg) [reply src/relax/op/op_common.h:257]: Good point, and this no longer tells the user which operation it was.  Updated.

**Comment by Lunderberg** [comment]
No problem, and thank you.  This isn't a high-priority PR to land, and can certainly wait until after the weekend.

**Review by tqchen — APPROVED** [review]

**Comment by tqchen** [comment]
thanks @Lunderberg should be good to go after ci

**Comment by Lunderberg** [comment]
@tqchen Thank you!  I've resolved the unit test whose failure was specific to this PR.

There's a few other CI failures, which look like they're triggered by a bug in `tvm.device('cuda').exist`.  If no GPUs are present, it raises an exception when it should return `False`.  This is an independent failure mode, and I've submitted https://github.com/apache/tvm/pull/16903 which resolves it.

**Pull request merged**
